src/lib/Component.coffee

```diff
@@ -358,17 +357,30 @@ class ProcessInput
     return @has.apply @, args
 
   # Returns true if a port has a complete stream in its input buffer.
-  hasStream: (port) ->
-    buffer = @buffer.get port
-    return false if buffer.length is 0
-    # check if we have everything until end of a complete stream
-    received = 0
-    for packet in buffer
-      if packet.type is 'openBracket'
-        ++received
-      else if packet.type is 'closeBracket'
-        --received
-    received is 0
+  hasStream: (args...) ->
+    args = ['in'] unless args.length
+    for port in args
+      portBrackets = []
+      hasData = false
+      validate = (ip) ->
+        if ip.type is 'openBracket'
+          portBrackets.push ip.data
+          return false
+        if ip.type is 'data'
+          # Data IP on its own is a valid stream
+          return true unless portBrackets.length
+          # Otherwise we need to check for complete stream
+          hasData = true
+          return false
+        if ip.type is 'closeBracket'
+          portBrackets.pop()
+          return false if portBrackets.length
+          return false unless hasData
```

Comment:
Yes, this is based on what we agreed as being a "stream" in the meetup last summer. Either:

* Data packet not surrounded by brackets
* Data packet(s) surrounded by matching brackets

So empty brackets are not considered a stream.